test/__init__.py

```diff
@@ -0,0 +1,166 @@
+
+
+def sagemaker_session(region_name=DEFAULT_REGION):  # type: (str) -> sagemaker.Session
+    return sagemaker.Session(boto3.Session(region_name=region_name))
+
+
+def default_bucket(session=None):  # type: (sagemaker.Session) -> str
+    session = session or sagemaker_session()
+    return session.default_bucket()
+
+
+def write_json(obj, path):  # type: (object, str) -> None
+    """Serialize ``obj`` as JSON in the ``path`` file.
+
+    Args:
+        obj (object): Object to be serialized
+        path (str): Path to JSON file
+    """
+    with open(path, 'w') as f:
+        json.dump(obj, f)
+
+
+def prepare(user_module, hyperparameters, channels, current_host='algo-1', hosts=None):
+    # type: (UserModule, dict, list, str, list) -> None
+    hosts = hosts or ['algo-1']
+
+    user_module.upload()
+
+    create_hyperparameters_config(hyperparameters, user_module.url)
+    create_resource_config(current_host, hosts)
+    create_input_data_config(channels)
+
+
+def hyperparameters(**kwargs):  # type: (...) -> dict
+    default_hyperparameters = DEFAULT_HYPERPARAMETERS.copy()
+
+    default_hyperparameters.update(kwargs)
+    return default_hyperparameters
+
+
+def create_resource_config(current_host, hosts):  # type: (str, list) -> None
+    write_json(dict(current_host=current_host, hosts=hosts), smc.environment.RESOURCE_CONFIG_PATH)
+
+
+def create_input_data_config(channels):  # type: (list) -> None
+    input_data_config = {channel.name: channel.config for channel in channels}
+
+    write_json(input_data_config, smc.environment.INPUT_DATA_CONFIG_FILE_PATH)
+
+
+def create_hyperparameters_config(hyperparameters, submit_dir, sagemaker_hyperparameters=None):
+    # type: (dict, str, dict) -> None
+    all_hyperparameters = {smc.environment.SUBMIT_DIR_PARAM: submit_dir}
+    all_hyperparameters.update(sagemaker_hyperparameters or DEFAULT_HYPERPARAMETERS.copy())
+    all_hyperparameters.update(hyperparameters)
+
+    write_json(all_hyperparameters, smc.environment.HYPERPARAMETERS_PATH)
+
+
+File = collections.namedtuple('File', ['name', 'content'])  # type: (str, str or list) -> File
+
+
+class UserModule(object):
+
+    def __init__(self, main_file, key=None, bucket=None, session=None):
+        # type: (File, str, str, sagemaker.Session) -> None
+        session = session or sagemaker_session()
+        self._s3 = session.boto_session.resource('s3')
+        self.bucket = bucket or default_bucket(session)
+        self.key = key or os.path.join('test', 'sagemaker-containers', str(time.time()), 'sourcedir.tar.gz')
+        self._files = [main_file]
+
+    def add_file(self, file):  # type: (File) -> UserModule
+        self._files.append(file)
+        return self
+
+    @property
+    def url(self):  # type: () -> str
+        return os.path.join('s3://', self.bucket, self.key)
+
+    def upload(self):  # type: () -> UserModule
```

Comment:
Can you reuse functionality from either the python SDK or from sagemaker-containers to do this part? I don't like having  complex test-only logic if we can avoid it.

Reply:
I tried before, but it can be done without changing the sdk, which would increase too much the scope at this moment.